scripts/install_prerequisites.sh

```diff
@@ -34,20 +49,32 @@ function temp_install_if_command_unknown {
 }
 
 function remove_temp_installs {
-  if [ "$APT_UNINSTALL" != "" ]; then
+  if [ -n "$APT_UNINSTALL" ]; then
       echo "Uninstalling packages used for bootstrapping: $APT_UNINSTALL"
       apt-get remove -y $APT_UNINSTALL && apt-get autoremove -y --purge
       unset APT_UNINSTALL
+      create_llvm_symlinks # uninstalling other compiler tools may have removed the symlinks
   fi
 }
 
-set -e
-trap remove_temp_installs EXIT
+read __errexit__ < <(echo $SHELLOPTS | egrep -o '(^|:)errexit(:|$)' || echo)
+function exit_gracefully {
+  remove_temp_installs
+  if [ -z "$__errexit__" ]; then set +e; fi
+}
+
+set -e && trap exit_gracefully EXIT
 this_file_dir=`dirname "$(readlink -f "${BASH_SOURCE[0]}")"`
 
 if [ ! -x "$(command -v cmake)" ]; then
-    apt-get update && apt-get install -y --no-install-recommends cmake
-    APT_UNINSTALL="$APT_UNINSTALL $2"
+  temp_install_if_command_unknown wget wget
+  wget https://github.com/Kitware/CMake/releases/download/v3.26.4/cmake-3.26.4-linux-$(uname -m).sh -O cmake-install.sh
```

Comment:
Shouldn't we use the minimum cmake version we require? 
https://github.com/NVIDIA/cuda-quantum/blob/03b1be551bc9ead4c3066784e190b8fde2be6779/CMakeLists.txt#L9

Reply:
Why?
I want/need to use 3.26 in the build for the Python wheels (you can see the 3.26 requirement for cmake in the toml file). I had originally updated the minimum required to 3.26, but then Eric had set it back because 3.26 is technically not required for building CUDA Quantum from the command line, see also https://github.com/NVIDIA/cuda-quantum/pull/423.

Reply:
My understanding is that python wheels are part of the project. Thus if we **need** cmake 3.26 to build them, then our minimum required version is not 3.22. 

So, either we create a workaround to build python wheels with cmake 3.22, or that is not our minimum required version. I would rather not go down the path where different parts of our project require different version of the same build dependency. However, if this is what we want, then at least add proper checks in the build system so that errors will be emitted with if someone tries to build wheels with a cmake version lower than 3.26.

Also, we need to check if we are not adding features only available to higher version of cmake (there is not easy way to do that other than using the minimum version). If our CI is not checking that, we are quite likely to run into problems. For example: In #1006, the only reason for the workaround when adding `libqpp` is because the feature that would make that straightforward is only available in cmake 3.25+.  (There might be other workarounds in the build system because of the version.) Had I not know this feature is not available in cmake 3.22, I could had wrongly added it, the CI would have passed, but an user with cmake 3.22 would not be able to correctly build from source.